src/Components/Facility/DischargeModal.tsx

```diff
@@ -169,6 +175,17 @@ const DischargeModal = ({
 
   const prescriptionActions = PrescriptionActions(consultationData.id ?? "");
 
+  const handleFacilitySelect = (selected: FacilityModel) => {
+    setFacility(selected ? selected : facility);
+    const { id, name } = selected;
+    const isExternal = id === -1;
+    setPreDischargeForm((prev) => ({
+      ...prev,
+      referred_to: isExternal ? null : name,
```

Comment:
We need to send 

- The facility ID in case of `referred_to` (you're sending the name here) 

- The facility name in case of `referred_to_external` (this part is done correctly).